rpc/transport.go

```diff
@@ -52,12 +52,35 @@ type Transport interface {
 	Close() error
 }
 
+// StreamTransportOption configures StreamTransport.  In general, both sides
+// of the connection must agree on options.
+type StreamTransportOption func(*StreamTransport)
```

Comment:
I'm actually not a huge fan of the option pattern for the downsides outlined in [this blog post](https://blog.golang.org/module-compatibility) (I provided some of the initial feedback that led to the recommendations in that blog post). Ultimately, I've found config structs to be easier to maintain.